test/e2e/clusterctl_upgrade.go

```diff
@@ -455,6 +480,28 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 			return n, nil
 		}, input.E2EConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Equal(expectedMachineCount), "Timed out waiting for all Machines to exist")
 
+		if skipMachinePoolMachines {
```

Comment:
Would it make sense to always run this code? (I assume this also works for MachinePools with Machines?)

Reply:
Yes this should also work, but it might be redundant since we already check for machines in the previous step. Do you think it's better to check anyways?

Reply:
This steps seems to wait for readiness, the previous one does not. I would prefer to have less branching here